mcs/class/corlib/Mono.Globalization.Unicode/SortKey.cs

```diff
@@ -77,6 +77,11 @@ internal SortKey (int lcid, string source, CompareOptions opt)
 			this.lcid = lcid;
 			this.source = source;
 			this.options = opt;
+			var source_length = source.Length;
+			var k = new byte [source_length];
+			for (int i = 0; i < source_length; ++i)
+				k [i] = (byte)source [i];
```

Comment:
shorter?
```csharp
this.key = new byte [source.Length];
for (int i = 0; i < source.Length; ++i)
  this.key [i] = (byte) source [i];
```

Reply:
I don't trust every `.` to be a reload from memory. Do you?
I agree it works and means the same thing of course.

Reply:
is it critical in terms of perf or raciness?

Reply:
No but I prefer to keep it faster when easy.

Reply:
Use the shorter version.